sympy/solvers/recurr.py

```diff
@@ -752,6 +752,10 @@ def rsolve(f, y, init=None):
 
     common = S.One
 
+    if not i_part.is_zero and not i_part.is_hypergeometric(n) and \
+       not (i_part.is_Add and all(map(lambda x: i_part.is_hypergeometric(n), f.expand().args))):
```

Comment:
Do you mean `all(map(lambda x: x.is_hypergeometric(n), f.expand().args))`?

Reply:
Exactly, that's my silly mistake.

Reply:
I believe your tests passed before this change which means that having that correct didn't matter. Can you add a tests which requires that this change be correct?

Reply:
In fact it should be `all(map(lambda x: x.is_hypergeometric(n), i_part.expand().args))`. I added tests for it.